### PR TITLE
IBP-3327-IBP-3328-UseMcodeAsReference

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Modified trial-settings-manager.js  script: remove filtering variables with property id 2670, this will allow showing of study 
variables option in study detail 
Modified trial-data-manager.js script: before submitting, use value.key instead of the id to capture selected breeding method. 
Removing parsing to int when codeTermId is used 
Added logic when deleting idCodeName paired variables 
Fix unit tests

### CODE REVIEW

File: src/main/java/com/efficio/fieldbook/service/api/FieldbookService.java [hunk 1]
```diff
@@ -176,6 +176,13 @@
 	 */
 	Map<String, String> getIdNamePairForRetrieveAndSave();
 
+	/**
+	 * Gets the id code name pair for retrieve and save.
+	 *
+	 * @return the id code name pair for retrieve and save
+	 */
+	Map<String, List<String>> getIdCodeNamePairForRetrieveAndSave();
```
Comment (abatac) [inline src/main/java/com/efficio/fieldbook/service/api/FieldbookService.java:184]: Rename to getIdCodeNamePairsForStudyConditions, update javadoc.

File: src/main/webapp/WEB-INF/static/js/angular/fieldbook-utils.js [hunk 2]
```diff
@@ -430,24 +435,42 @@
                         var currentVal = $scope.valuecontainer[$scope.targetkey];
 
 						// lets fix current val if its an object so that it only contains the id
-						if (typeof currentVal !== 'undefined' && currentVal !== null && typeof currentVal.id !== 'undefined' && currentVal.id) {
-							currentVal = currentVal.id;
+						if($scope.isBreedingMethod){
+							if (typeof currentVal !== 'undefined' && currentVal !== null && typeof currentVal.key !== 'undefined' && currentVal.key) {
```
Comment (abatac) [inline src/main/webapp/WEB-INF/static/js/angular/fieldbook-utils.js:439]: Can this be simplified into this?: (currentVal && currentVal.key)

File: src/main/webapp/WEB-INF/static/js/trialmanager/trial-settings-manager.js [hunk 3]
```diff
@@ -203,17 +203,6 @@ window.TrialSettingsManager = (function() {
 
 		// Currently we only have special handling for management details (1) and selection strategy (6)
 		switch (group) {
-			case 1805:
```
Comment (abatac) [inline src/main/webapp/WEB-INF/static/js/trialmanager/trial-settings-manager.js:206]: Does this only affect the the Study Details form?

**Review by abatac — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/static/js/trialmanager/trial-settings-manager.js (hunk 3 above)
Comment (gelli-leafnode) [reply src/main/webapp/WEB-INF/static/js/trialmanager/trial-settings-manager.js:206]: Hi @abatac , 
Compared in master datasetimport and design import performs the same (It shows study_bm_code). 

Thanks,
Gelli

**Pull request merged**